ibis/expr/types.py

```diff
@@ -1434,7 +1446,10 @@ def literal(value, type=None):
     if hasattr(value, 'op') and isinstance(value.op(), Literal):
         return value
 
-    dtype = infer_literal_type(value)
+    if value is null:
```

Comment:
Why is `null` a callable? To ensure immutability of `_NULL`?

Should I register a to `dt.infer` with `FunctionType`?